src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -310,8 +312,16 @@ def set_assessment_substatus_json(self, status=Constants.STATUS_TRANSITIONING, c
         # Wrap assessment summary into assessment substatus
         self.__assessment_substatus_json = self.__new_substatus_json_for_operation(Constants.PATCH_ASSESSMENT_SUMMARY, status, code, json.dumps(self.__assessment_summary_json))
 
-        # Update status on disk
-        self.__write_status_file()
+        # Update status complete on disk
+        self.__write_status_complete_file()
+
+        # Write truncated status on disk, apply truncation logic here
+        status_json_payload = self.__read_status_file_raw_data(self.status_complete_file_path)
+        # print('what is status json_payload', status_json_payload)
+        self.env_layer.file_system.write_with_retry_using_temp_file(self.status_file_path,
+                                                                    '[{0}]'.format(json.dumps(status_json_payload)),
+                                                                    mode='w+')
```

Comment:
Not leaving the same comment everywhere but avoid multi-lines where possible. You can stop the IDE from prompting you on this. The IDE prompts are based on being good to folk who are trying to edit code in the terminal later (this does not apply to us).